default_executor/src/lib.rs

```diff
@@ -190,12 +191,14 @@ pub fn validate_instruction(
 ) -> Result {
     let mut executor = Executor::new(block_height);
     executor.visit_instruction(&authority, &instruction);
+    core::mem::forget(instruction);
     executor.verdict
 }
 
 #[entrypoint]
 pub fn validate_query(authority: AccountId, query: QueryBox, block_height: u64) -> Result {
     let mut executor = Executor::new(block_height);
     executor.visit_query(&authority, &query);
+    core::mem::forget(query);
```

Comment:
I don't think it would be a bad idea to inject these `forget`s by the `entrypoint`?